src/SignalR/clients/csharp/Client.Core/src/HubConnectionExtensions.StreamAsChannelAsync.cs

```diff
@@ -284,13 +284,13 @@ async Task RunChannel()
             {
                 try
                 {
-                    while (await inputChannel.WaitToReadAsync())
+                    while (await inputChannel.WaitToReadAsync(cancellationToken))
```

Comment:
This isn't right. We want to cancel the initial call with the token, not the enumeration.